keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,295 @@
+        level = self._random_generator.random_uniform(
+            shape=shape, minval=0.1, maxval=level, dtype=tf.float32
+        )
+        return tf.cast((level) * maxval / 10.0, dtype)
+
+    def _loop_on_depth(self, depth_level, image_aug):
+        op_index = self._random_generator.random_uniform(
+            shape=(), minval=0, maxval=8, dtype=tf.int32
+        )
+        image_aug = self._apply_op(image_aug, op_index)
+        depth_level += 1
+        return depth_level, image_aug
+
+    def _loop_on_width(self, image, chain_mixing_weights, curr_chain, result):
+        image_aug = tf.identity(image)
+        depth = tf.cond(
+            tf.greater(self.depth, 0),
+            lambda: self.depth,
+            lambda: self._random_generator.random_uniform(
+                shape=(), minval=1, maxval=3, dtype=tf.int32
+            ),
+        )
+
+        depth_level = tf.constant([0], dtype=tf.int32)
+        depth_level, image_aug = tf.while_loop(
+            lambda depth_level, image_aug: tf.less(depth_level, depth),
+            self._loop_on_depth,
+            [depth_level, image_aug],
+        )
+        result += tf.gather(chain_mixing_weights, curr_chain) * image_aug
+        curr_chain += 1
+        return image, chain_mixing_weights, curr_chain, result
+
+    def _auto_contrast(self, image):
+        return layers.AutoContrast(self.value_range)(image)
+
+    def _equalize(self, image):
+        shape = image.shape
+        image = layers.Equalization(self.value_range)(image)
+        image.set_shape(shape)
+        return image
+
+    def _posterize(self, image):
+        image = preprocessing.transform_value_range(
+            images=image,
+            original_range=self.value_range,
+            target_range=[0, 255],
+        )
+
+        bits = self._sample_level(self.severity, 3, tf.int32)
+        shift = tf.cast(4 - bits + 1, tf.uint8)
+        image = tf.cast(image, tf.uint8)
+        image = tf.bitwise.left_shift(tf.bitwise.right_shift(image, shift), shift)
+        image = tf.cast(image, self.compute_dtype)
+        return preprocessing.transform_value_range(
+            images=image,
+            original_range=[0, 255],
+            target_range=self.value_range,
+        )
+
+    def _rotate(self, image):
+        angle = self._sample_level(self.severity, 30, tf.float32, [1])
+        shape = tf.cast(tf.shape(image), tf.float32)
+
+        return preprocessing.transform(
+            tf.expand_dims(image, 0),
+            preprocessing.get_rotation_matrix(angle, shape[0], shape[1]),
+        )[0]
+
+    def _solarize(self, image):
+        threshold = tf.cast(
+            self._sample_level(self.severity, 255, tf.int32), tf.float32
+        )
+
+        image = preprocessing.transform_value_range(
+            image, original_range=self.value_range, target_range=(0, 255)
+        )
+        result = tf.clip_by_value(image, 0, 255)
+        result = tf.where(result < threshold, result, 255 - result)
```

Comment:
same question here, can we use the solarize layer here?

Reply:
Same reason as posterize.